azkaban-common/src/main/java/azkaban/executor/ExecutorApiGateway.java

```diff
@@ -75,7 +114,10 @@ public ExecutorApiGateway(final ExecutorApiClient apiClient) {
           .valueOf(executionId)));
       paramList.add(new Pair<>(ConnectorParams.USER_PARAM, user));
 
-      return callForJsonObjectMap(host, port, "/executor", paramList);
+      // Ideally we should throw an exception if executionId is null but some existing code
+      // (updateExecutions()) expects to call this method with a null executionId.
+      String executionPath = (executionId == null) ? null : createExecutionPath(executionId);
```

Comment:
Should this be "/executor" (or an equivalent const) instead of `null` if executionId is null?

Reply:
Thanks for reporting this, you are spot-on about the root-cause. Code for the fix is nearly done,  will be ready for a PR later today (after some tests).